tmt/steps/report/reportportal.py

```diff
@@ -91,77 +139,222 @@ class ReportReportPortal(tmt.steps.report.ReportPlugin[ReportReportPortalData]):
 
     def handle_response(self, response: requests.Response) -> None:
         """
-        Check the server response and raise an exception if needed.
+        Check the endpoint response and raise an exception if needed.
         """
 
         if not response.ok:
             raise tmt.utils.ReportError(
                 f"Received non-ok status code from ReportPortal: {response.text}")
 
-        self.debug("Response code from the server", response.status_code)
-        self.debug("Message from the server", response.text)
+        self.debug("Response code from the endpoint", str(response.status_code))
+        self.debug("Message from the endpoint", str(response.text))
 
     def go(self) -> None:
         """
-        Report test results to the server
+        Report test results to the endpoint
 
         Create a ReportPortal launch and its test items,
         fill it with all parts needed and report the logs.
         """
 
+        # TODO: ? why it did not closed :
+        #           tmt run discover provision -h connect -g 10.0.185.29 -u root
+        #               -p fo0m4nchU  prepare report -h reportportal --suite-per-plan
+        #               --defect-type idle -vvv finish tests -n .
+        #       * test uploading to idle tests
+        #       * resolve the problem with mypy
+        #       * replace rp_uuid with an universal structure for tmt plugin data
+        #       * check the problem with --upload-to-suite functonality
+        #       * check the param per plan, and do the matching when uploading (launch_uuid)
+        #       * check the defect_type, idle functionality
+        #       * upload documentation (help and spec)
+        #       * add the tests for new features
+        #       * add uploading files
+        #       * edit schemas
+
+        #       * check optionality in arguments
+        #       * restrict combinations + set warning
+        #                   - forbid rerun && launch-per-plan
+        #                   - forbid upload-to-suite && launch-per-plan
+        #       * try read a launch_uuid at first plan (self.step.plan.report.launch_uuid)
+
         super().go()
 
-        server = self.data.url
-        if not server:
-            raise tmt.utils.ReportError("No ReportPortal server url provided.")
-        server = server.rstrip("/")
+        endpoint = self.get("url")
```

Comment:
I suspect this is a combination of the PR being fairly old and changes that happened in the meantime, rather than the actual change you'd like to make. But, if that's an actual change, I'd strongly advise not to use `get()`, `self.data` should be used instead. `self.data.foo` is properly annotated while `self.get('foo')` returns `Any`, lowering guarantees we get from type annotation linters.